configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyConfiguration.java

```diff
@@ -172,6 +177,20 @@ public boolean wasKeyProvided(@NotNull final String key) {
         return getRaw(keys::contains);
     }
 
+    @NotNull
+    public Map<String, String> getRawKeyValueMap(@NotNull final Set<Property> properties) {
+        Map<String, String> rawMap = new HashMap<>();
+        for (Property property : properties) {
+            if (!wasKeyProvided(property.getKey())) {
+                continue;
+            }
+
+            String rawValue = getRaw(property).orElse("");
```

Comment:
I'd rather we did `.ifPresent(value -> rawMap.put(property.getKey(), value))`. It's probably functionally the same since if the key was provided there should always be a raw value present but this at least prevents the empty string from being incorrectly returned. I'd even be fine doing `getOrThrow(new Exception("Something has gone wrong with the properties"))` as something has truly gone wrong if you don't have a raw but the key was provided. I just think doing the .orElse("") is asking for trouble down the line if it does become an issue.

Reply:
I like the ifPresent suggestion.  I suppose the current code might be looking for the case where a user passes in a property key but doesn't provide a value (ie. --blackduck.username=), but Spring appears to error out when that happens.